h/migrations/versions/5ed9c8c105f6_add_authority_provided_id_field_to_group.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+"""
+Add authority_provided_id field, index to group
+
+This will allow clients of the API, in certain circumstances, to set a
+unique-to-authority identifier on a group, versus having to use the
+service-generated ``pubid``. This ``authority_provided_id`` can be used
+to perform certain actions on the group without having to know the
+``pubid`` beforehand.
+
+Whereas the ``pubid`` is service-owned and globally unique, this
+``authority_provided_id`` is unique per authority and owned by the
+caller/client. i.e. authority-bound clients may set their own unique
+IDs.
+
+``authority_provided_id`` must be unique per authority; ergo the
+unique-enforced index
+"""
+from __future__ import unicode_literals
+from __future__ import absolute_import
+from __future__ import division
+
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = "5ed9c8c105f6"
+down_revision = "5d256923d642"
+
+
+def upgrade():
+    op.add_column('group', sa.Column('authority_provided_id', sa.UnicodeText(), nullable=True))
+    op.execute('COMMIT')
+    op.create_index(op.f('ix__group__groupid'),
+                    'group',
+                    ['authority_provided_id', 'authority'],
```

Comment:
Assuming the order here is used to order index entries, I think it would make sense to put the authority first. This would, I think, enable a more efficient query if we ever needed to iterate over provided IDs under a given authority.